### PR TITLE
Fix CC smiles

### PR DESCRIPTION
**Pull request by xiaoruiDong**
### Motivation or Problem
This PR tries to fix issue #2536, where the SMILES of bicarbon (CC) is not correctly generated.

### Description of Changes
This PR forces the setting of no implicit Hs when converting the RMG Molecule to an RDKit Mol object. This is viable since all hydrogens in the RMG Molecules are defined explicitly.

### Testing
Add a few unit tests to test the molecule of CC with various multiplicities. I comment out the case with a quadruple bond, as it is very likely the installed RDKit version doesn't have support for output smiles of quadruple bond (which is introduced in sometime 2022)

<del> There is an InChI error raised due to this change, and I am investigating it.
https://github.com/xiaoruiDong/RMG-Py/actions/runs/6238513699/job/16934497099

### CODE REVIEW

**Comment by xiaoruiDong** [comment]
This seems to be an RMG inchi parser problem. The following results are from the current `main`:

```
Molecule().from_augmented_inchi('InChI=1S/C5H6O/c6-5-3-1-2-4-5/h1-3,5H,4H2/u1,6')
```
Got InvalidAdjacencyListError

```
Molecule().from_augmented_inchi('InChI=1S/C5H6O/c6-5-3-1-2-4-5/h1-3,5H,4H2/u2,6').to_augmented_inchi()
>>> 'InChI=1S/C5H6O/c6-5-3-1-2-4-5/h1-3,5H,4H2/u1,6'
```

```
Molecule().from_augmented_inchi('InChI=1S/C5H6O/c6-5-3-1-2-4-5/h1-3,5H,4H2/u2,6').to_smiles()
>>> '[O]C1C=C[CH]C1'
Molecule().from_smiles(smi).to_augmented_inchi()
>>> 'InChI=1S/C5H6O/c6-5-3-1-2-4-5/h1-3,5H,4H2/u1,6'
[mol.to_augmented_inchi() for mol in Molecule().from_smiles(smi).generate_resonance_structures()]
>>> ['InChI=1S/C5H6O/c6-5-3-1-2-4-5/h1-3,5H,4H2/u1,6',
     'InChI=1S/C5H6O/c6-5-3-1-2-4-5/h1-3,5H,4H2/u3,6']

**Review by JacksonBurns — APPROVED** [review]
Thanks for the quick patch @xiaoruiDong! This looks thorough and quite reasonable to me, and the regression tests indicate that this isn't impacting anything. I think we should merge, but let's get additional comments from @hwpang as well.

File: rmgpy/molecule/inchi.py [hunk 1]
```diff
@@ -602,13 +602,13 @@ def create_augmented_layers(mol):
     else:
         molcopy = mol.copy(deep=True)
 
+        rdkitmol = to_rdkit_mol(molcopy, remove_h=True)
+        _, auxinfo = Chem.MolToInchiAndAuxInfo(rdkitmol, options='-SNon')  # suppress stereo warnings
+
```
Comment (rwest) [inline rmgpy/molecule/inchi.py:607]: Is Codecov correct that lines #L605 - L606 were not covered by tests ? Does that mean this whole method is not tested? (or is codecov not right?)

**Review by rwest — COMMENTED** [review]
Made a couple comments. In general looks like a good fix. Thanks
I'm not sure how to interpret the Codecov annotations - I assume it's being tested?

**Comment by bjkreitz** [comment]
Thanks, this seems to work properly. I tested it for the different CC structures and also some other species with unpaired electrons at the carbon atom such as CCH3. RMG returned the correct SMILES string.

**Review by bjkreitz — APPROVED** [review]

File: rmgpy/molecule/converter.py [hunk 2]
```diff
@@ -116,6 +116,8 @@ def to_rdkit_mol(mol, remove_h=True, return_mapping=False, sanitize=True, save_o
         for label, ind_list in label_dict.items():
             for ind in ind_list:
                 Chem.SetSupplementalSmilesLabel(rdkitmol.GetAtomWithIdx(ind), label)
+    [atom.SetNoImplicit(True) for atom in rdkitmol.GetAtoms()
```
[the remark this answers is not in the document]
Comment (xiaoruiDong) [reply rmgpy/molecule/converter.py]: Thanks for the suggestion! With some Google searches, the use of list comprehension here is less Pythonic, as my purpose is not to create a list. I've changed this as suggested.

File: rmgpy/molecule/inchi.py (hunk 1 above)
Comment (xiaoruiDong) [reply rmgpy/molecule/inchi.py:607]: This is something weird. The failures in the unit tests testing this function inspired my change here. I looked at the [codecov coverage profile](https://app.codecov.io/gh/ReactionMechanismGenerator/RMG-Py/pull/2545/blob/rmgpy/molecule/inchi.py) and found it reported 0% coverage of the whole inchi.py, but RMG actually has [unit tests](https://github.com/ReactionMechanismGenerator/RMG-Py/blob/main/test/rmgpy/molecule/inchiTest.py) for it. @JacksonBurns any thoughts?

Comment (JacksonBurns) [reply rmgpy/molecule/inchi.py:607]: So this test is definitely running (you can search for it in the CI logs) and the new code (and this whole file) are being run, so CodeCov is just wrong. From brief googling I can't find a solution, so I will debug this separately, but this PR should be fine.

**Review by JacksonBurns — APPROVED** [review]
The new unit tests cover this edge case well, and the solution is solid. CodeCov is incorrectly reporting the result, which is a problem for another time. This PR LGTM

**Review by bjkreitz — APPROVED** [review]
